src/ed_py/entity.py

```diff
@@ -0,0 +1,308 @@
+            uuid=self.uuid, etype=self.etype, frame=self.pose, pp=self._person_properties
+        )
+
+
+class PersonProperties(object):
+    def __init__(
+        self,
+        name: str,
+        age: int,
+        emotion: str,
+        gender: str,
+        gender_confidence: float,
+        pointing_pose,
+        posture: str,
+        reliability,
+        shirt_colors: List[str],
+        tags,
+        tagnames,
+        velocity,
+        parent_entity,
+    ):
+        # ToDo: "In the legacy message definition a field named tagnames was defined but never used. Only tags is used.
+        #  Once the message definition in people_recognition is cleaned up then tagnames should be removed completely."
+        """
+        Container for several properties related to a person
+
+        :param name: the person's name. This is separate from the entity, which is unique while this doesn't have to be
+        :param age: Estimated age of the person
+        :param emotion: Indicating the emotion
+        :param gender: Predicted gender of the person
+        :param gender_confidence: Confidence of the classifier in the gender above.
+        :param pointing_pose: In which direction is the person pointing
+        :param posture: String with a value like 'sitting', 'laying', 'standing' etc.
+        :param reliability:  ToDO
+        :param shirt_colors: list of 3 shirt colors, sorted from most dominant to less dominant
+        :param tags: Other tags
+        :param tagnames: Other tagnames
+        :param velocity: Velocity with which the person in moving
+        :param parent_entity: The Entity that these properties belong to
+        """
+        self._name = name
+        self.age = age
+        self.emotion = emotion
+        self.gender = gender
+        self.gender_confidence = gender_confidence
+        self.pointing_pose = pointing_pose
+        self.posture = posture
+        self.reliability = reliability
+        self.shirt_colors = shirt_colors
+        self.tags = tags
+        self.tagnames = tagnames
+        self.velocity = velocity
+
+        self._parent_entity = parent_entity
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @name.setter
+    def name(self, value: str):
+        rospy.loginfo("Changing {}'s name to {}".format(self._parent_entity.id, value))
+        self._name = value
+
+    def __repr__(self):
+        return "PersonProperties(age='{age}', gender='{g}', gender_confidence={gc}, shirt_colors={sc})".format(
+            age=self.age, g=self.gender, gc=self.gender_confidence, sc=self.shirt_colors
+        )
+
+
+def from_entity_info(e: EntityInfo) -> Entity:
+    """
+    Converts ed_msgs.msg.EntityInfo to an Entity
+
+    :param e: ed_msgs.msg.EntityInfo
+    :return: Entity
+    """
+    assert isinstance(e, EntityInfo)
+    identifier = e.id
+    object_type = e.type
+    frame_id = "map"  # ED has all poses in map
+    # ToDo: Change pose in msg definition to PoseStamped
```

Comment:
Can/should this be done now?

Reply:
Related to review in package.xml. This would require a lot of changes in other repos